core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -250,6 +250,9 @@ public void run() {
       loadBalancer.shutdown();
       loadBalancer = null;
```

Comment:
I don't like this code in `ManagedChannelImpl`:
```
LbHelperImpl helper = new LbHelperImpl(nameResolver);
helper.lb = loadBalancerFactory.newLoadBalancer(helper);
this.loadBalancer = helper.lb;
```

Because `ManagedChannelImpl` does not own `loadBalancerFactory`, it does not know the behavior of `loadBalancerFactory.newLoadBalancer()`, which may just always recycle the old loadBalancer and restart it so that `loadBalancer == lb` is always true.